include/eve/arch/riscv/as_register.hpp

```diff
@@ -0,0 +1,299 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/arch/riscv/predef.hpp>
+#include <eve/arch/riscv/rvv_utils.hpp>
+#include <eve/traits/as_integer.hpp>
+
+#include <type_traits>
+#if defined(EVE_INCLUDE_RISCV_HEADER)
+
+namespace eve
+{
+template<typename T> struct logical;
+}
+
+#  if defined(EVE_RETURN_M)
+#    error EVE_RETURN_M already defined
+#  endif
+
+#  if defined(EVE_RETURN_MF)
+#    error EVE_RETURN_MF already defined
+#  endif
+
+#  define EVE_RETURN_MF(REG, LMUL)                                                                 \
+    do {                                                                                           \
+      using type = REG __attribute__((riscv_rvv_vector_bits(__riscv_v_fixed_vlen / LMUL)));        \
+      return type {};                                                                              \
+    }                                                                                              \
+    while( 0 )
+
+#  define EVE_RETURN_M(REG, LMUL)                                                                  \
+    do {                                                                                           \
```

Comment:
the macro is used in a if consetxpr context, why not putting the { } in the if constexpr and just have the macro be the using + return. The do while(0) here is a bit overkill

Reply:
It was my suggetion - I liked the semicolon. If Joel doesn't like it - do the brackets

Reply:
Done